pg_auth_mon.c

```diff
@@ -130,18 +139,61 @@ fai_shmem_startup(void)
 static void
 fai_shmem_shutdown(int code, Datum arg)
 {
+
+	last_log_timestamp = NULL;
 	auth_mon_ht = NULL;
 
 	return;
 }
 
+/*
+ * Write content of the `pg_auth_mon` to regular Postgres log to make it searchable later. 
+ * 
+ * The caller must have the auth_mon_lock in the LW_EXCLUSIVE mode.
+ */
+static void log_pg_auth_mon_data(){
+
+	HASH_SEQ_STATUS status;
+	auth_mon_rec *entry;
+	const char *last_successful_login_at;
+	const char *last_failed_attempt_at;
+
+	/* 
+	 * log at most once in an hour 
+	*/
+	int waittime = 1000 * 60 * 60;
+	TimestampTz now = GetCurrentTimestamp();
+
```

Comment:
I am a little unsure about this, 
1. should we do it once per day?
2. shouldn't we do it at the time of shutdown too?